qubes/vm/dispvm.py

```diff
@@ -187,11 +187,37 @@ def __init__(self, app, xml, *args, **kwargs):
             self.features.update(template.features)
             self.tags.update(template.tags)
 
+    def is_domain_preloaded(self):
+        preload_dispvm = (
+            self.features.check_with_template("preload-dispvm", None)
+        )
+        if not preload_dispvm:
+            return False
+        preload_dispvm = preload_dispvm.split(" ")
+        if self.name not in preload_dispvm:
+            return False
+        return True
+
+
     @qubes.events.handler("domain-load")
     def on_domain_loaded(self, event):
         """When domain is loaded assert that this vm has a template."""  # pylint: disable=unused-argument
         assert self.template
 
+    @qubes.events.handler("domain-started")
+    def on_domain_started(self, event, **kwargs):
+        """Pause preloaded domains as soon as they start."""
+        if self.is_domain_preloaded():
+            self.pause()
```

Comment:
This is easy thing to do, but requires testing if it isn't too early. Some services (especially gui-agent) may still be starting.
There is an `qubes.WaitForSession` service that may help (ensure to use async handler to not block qubesd while waiting on it).
And also, the event is called `domain-start`.

Reply:
> What is preloaded dispvm has `gui` feature disabled in case the user is using the disposable to be qubes-builder-dvm for example, that doesn't require a GUI? What can be done in this case?

Very good question. In that case I guess start even is all we have. Maybe with some (configurable?) delay?